modules/backend/traits/ModelOptions.php

```diff
@@ -0,0 +1,92 @@
+<?php namespace Backend\Traits;
+
+/**
+ * Model Options Trait
+ * Implements helper methods useful to extract the options from a given field
+ * that is used in the List and Form widgets.
+ */
+
+trait ModelOptions
+{
+    /**
+     * Looks at the model for defined options.
+     *
+     * @param $field
+     * @param $fieldOptions
+     * @return mixed
+     */
+    protected function getOptionsFromModel($field, $fieldOptions, $fieldName = '')
```

Comment:
Can you replace references of `$field` to `$attribute` and `$fieldOptions` to `$options` to reflect the fact that this is generic now?

Reply:
Ok, let me see what can be done.

Reply:
I'm trying to figure out what name should I give to make it more generic to widgets. I'm stuck between elements and attributes. The latter seems to be something specific to eloquent models in general and there's also a variable called attribute used in the code. Do you have any suggestion?

Reply:
`$fieldOptions` can just be `$options`, however the more I look at this the less I'm certain that it even works for the list type given `->resolveModelAttribute($this->model)` being called on a Model instance and `$field->fieldName` being used in multiple spots when the lists widget isn't sending a field instance it's sending a model instance. Please look into this further and make it more generic / better support both uses.